File: leetcode/src/main/java/array/IntersectionOfMultipleArrays.java
package array;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

import static org.junit.jupiter.api.Assertions.*;

/**
 * https://leetcode.com/problems/intersection-of-multiple-arrays/
 *
 * Tìm phần tử chung của n mảng
 */
public class IntersectionOfMultipleArrays {

    public List<Integer> intersection(int[][] nums) {
        Set<Integer> result = Arrays.stream(nums[0]).boxed().collect(Collectors.toSet());
        for (int i = 1; i < nums.length; i++) {
            Set<Integer> another = Arrays.stream(nums[i]).boxed().collect(Collectors.toSet());
            result.retainAll(another);
        }
        return result.stream().sorted().toList();
    }

    @Test
    public void test() {
        assertIterableEquals(List.of(3, 4), intersection(new int[][]{new int[]{3, 1, 2, 4, 5}, new int[]{1, 2, 3, 4}, new int[]{3, 4, 5, 6}}));
    }

}

File: py-algorithm/arrays/intersection_of_multiple_arrays.py
import unittest
from typing import List

"""
    https://leetcode.com/problems/intersection-of-multiple-arrays/
    
    Tìm phần tử chung của n mảng
"""


class Solution(unittest.TestCase):

    @staticmethod
    def intersection(nums: List[List[int]]) -> List[int]:
        result = set(nums[0])
        for i in range(1, len(nums)):
            # result &= set(nums[i])
            result.intersection_update(nums[i])

        return sorted(list(result))

    def test(self):
        self.assertEqual([3, 4], self.intersection([[3, 1, 2, 4, 5], [1, 2, 3, 4], [3, 4, 5, 6]]))


if __name__ == '__main__':
    unittest.main()
